moveit_ros/moveit_servo/src/servo_parameters.cpp

```diff
@@ -230,6 +230,13 @@ void ServoParameters::declare(const std::string& ns,
   node_parameters->declare_parameter(
       ns + ".hard_stop_singularity_threshold", ParameterValue{ parameters.hard_stop_singularity_threshold },
       ParameterDescriptorBuilder{}.type(PARAMETER_DOUBLE).description("Stop when the condition number hits this"));
+  // Default approaching_stop_singularity_threshold to hard_stop_singularity_threshold
+  node_parameters->declare_parameter(
+      ns + ".approaching_stop_singularity_threshold",
+      ParameterValue{ parameters.approaching_stop_singularity_threshold },
+      ParameterDescriptorBuilder{}
+          .type(PARAMETER_DOUBLE)
+          .description("Stop when the condition number hits this and we are continuing to approach singularity"));
```

Comment:
@henrygerardmoore  can you add this simple check around L355. I think you'll see what I mean:

```
if (parameters.hard_stop_singularity_threshold <= parameters.approaching_stop_singularity_threshold)
{
  ...
}
```

Because `hard_stop_singularity_threshold` should always be > `approaching_stop_singularity_threshold`